### PR TITLE
Creating .gitignore

### LINKED ISSUE #2: .gitignore file not present
In order to make commits easier, specially in modern text editors and IDEs, it is interesting to keep the working tree clean even after the project has been built. Currently it's really messy (a lot of modifications are not desirable to push to the remote repo).

I will create a PR with a proposed .gitignore and link it with this issue soon.

Screenshots:

<details>
  <summary>VSCode Example</summary>
  
vscode-example
  
</details>

<details>
  <summary>Command `git status` Example</summary>
  
terminal-example
  
</details>

### PR DESCRIPTION
**Pull request by marcuscastelo**
Fixes #2

I've analyzed which files were created in each step of the build process, putting them in the proposed .gitignore file.
One problem I've found is that the file `build-aux/ltmain.sh` needs to exist in the original repo (not built), but it is modified during the build process. Because of that, it is not possible to include it in the .gitignore file and it is (currently) hanging as a modified file (after build process).

### CODE REVIEW

**Review by gsasouza — APPROVED** [review]
LGTM

**Comment by jufantozzi** [comment]
Did you test to see if the build initially works without the file? It looks like a temporary file that was accidentally commited. If that's the case, I'd suggest we also remove that file.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+autom4te.cache/
+
+m4/libtool.m4
+m4/ltversion.m4
+m4/lt~obsolete.m4
+m4/ltoptions.m4
+m4/ltsugar.m4
+
+# This file is modified during autogen.sh, not sure if it should be added to .gitignore
+build-aux/ltmain.sh
```
Comment (jufantozzi) [inline .gitignore:14]: This is most likely a left over. Code can be sucessfully compiled without this file. We should remove it and keep it in the gitignore. Just clean up the comment

**Review by jufantozzi — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (marcuscastelo) [reply .gitignore:14]: Interesting, I remember testing it in a fresh environment (clonned repo again) and it didn't work.
But maybe I got confused and tried it in the wrong folder. I can't test it right now, but maybe tomorrow I'll try it again.

**Review by lucasvianav — APPROVED** [review]

**Comment by marcuscastelo** [comment]
Tested again, the project builds without any errors deleting `build-aux/ltmain.sh`. I suppose I ended up deleting some other files the first time I tried to delete it.

**Review by Float07 — APPROVED** [review]

**Comment by marcuscastelo** [comment]
We have already 3 approvals, so I am going to merge the PR into `develop`.

**Pull request merged**